src/implementation/Client/GRPCClient/state.ts

```diff
@@ -180,4 +182,33 @@ export default class GRPCClientState implements IClientState {
       });
     });
   }
+
+  async query(storeName: string, query: StateQueryType): Promise<StateQueryResponseType> {
```

Comment:
Please, do not bundle the query API changes in this PR. This can be merged independently. BTW, there is a breaking change in this API already in 1.6, please sync with @dmitsh

Reply:
Personally, rather than extracting this I would prefer to merge it now and fix it afterwards in a new PR. Bit difficult to get it out now :) thoughts?

Reply:
Well. If the code is in separate files, separating is not that hard. Also, if you have it in separate commits, you can remove the commit via a rebase and then cherry-pick it in a new branch.

Reply:
Removed it